backend/src/Designer/Infrastructure/AnsattPorten/AnsattPortenExtensions.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Configuration;
+using Altinn.Studio.Designer.Constants;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Altinn.Studio.Designer.Infrastructure.AnsattPorten;
+
+public static class AnsattPortenExtensions
+{
+    public static IServiceCollection AddAnsattPortenAuthenticationAndAuthorization(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddAnsattPortenAuthentication(configuration);
+        services.AddAnsattPortenAuthorization(configuration);
+        return services;
+    }
+    private static IServiceCollection AddAnsattPortenAuthentication(this IServiceCollection services, IConfiguration configuration)
+    {
+        bool ansattPortenFeatureFlag = configuration.GetSection($"FeatureManagement:{StudioFeatureFlags.AnsattPorten}").Get<bool>();
+        if (!ansattPortenFeatureFlag)
+        {
+            return services;
+        }
+
+        AnsattPortenLoginSettings oidcSettings = configuration.GetSection(nameof(AnsattPortenLoginSettings)).Get<AnsattPortenLoginSettings>();
+
+        services
+            .AddAuthentication(AnsattPortenConstants.AnsattpotenCookiesAuthenticationScheme)
+            .AddCookie(AnsattPortenConstants.AnsattpotenCookiesAuthenticationScheme, options =>
+            {
+                options.Cookie.HttpOnly = true;
+                options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
+                // options.Cookie.SameSite = SameSiteMode.Lax;
```

Comment:
is `strict` the default value when the option is not provided? 🤔 Should we just remove this out-comment code? I belive that we do not want the `Lax` settings that allow cross-domain?